src/software/ai/hl/stp/tactic/move_tactic.h

```diff
@@ -1,7 +1,59 @@
 #pragma once
 
-#include "software/ai/hl/stp/action/move_action.h"
+#include "software/ai/hl/stp/action/move_action.h"  // TODO (#1888): remove this dependency
 #include "software/ai/hl/stp/tactic/tactic.h"
+#include "software/ai/intent/move_intent.h"
+
+struct MoveFSM
```

Comment:
I might be missing a trick, but does this need to be publicly exposed? Seems like it's just an implementation detail for this tactic. If so, might be worth defining in an anonymous namespace.

Reply:
The FSM does need to be publicly exposed for other Tactics to use this FSM as a sub-FSM. I did consider putting just the FSM in a separate file, but I wasn't sure that'd be any better